src/Planners/src/SwingFootPlanner.cpp

```diff
@@ -123,130 +129,141 @@ bool SwingFootPlanner::initialize(std::weak_ptr<const IParametersHandler> handle
-    // - If the contact is not active we have to check that:
-    //    1. the final orientation may change still the error (in the tangent space) between the
-    //       new orientation and the current one should be parallel to the current velocity and
-    //       acceleration vectors. This is required to keep the SO3Planner problem still treatable
-    //       online. This check is not done here since the SO3Planner will complain in case of
-    //       issues.
-    //    2. the swing foot trajectory duration did not change
+    //// Case 2: m_contactList is NOT empty (no need to check it given the return true) and a
+    ///          contact is currently active. In this case we can update the contact list if the new
+    ///          contact list has an active contact at the current time instant that has the same
+    ///          POSE of the current
 
-    // Here the case where the contact is active
+    // get the contact active at the given time
+    auto activeContactNewList = contactList.getActiveContact(m_currentTrajectoryTime);
+    auto activeContact = m_contactList.getActiveContact(m_currentTrajectoryTime);
 
-    // get the contact ative at the given time
-    auto newContactAtCurrentTime = contactList.getPresentContact(m_currentTrajectoryTime);
-
-    // if the contact is active and the new contact list has an active contact at the give time
-    // instant with the same position and orientation
-    if (m_state.isInContact && newContactAtCurrentTime != contactList.cend()
-        && newContactAtCurrentTime->isContactActive(m_currentTrajectoryTime)
-        && (*newContactAtCurrentTime) == (*m_currentContactPtr))
+    constexpr double tolerance = 1e-6;
+    if (activeContact != m_contactList.cend() // i.e., the original list contains an active contact
+        && activeContactNewList != contactList.cend() // i.e., the new list contains an active
+                                                      // contact
+    )
     {
-        m_contactList = contactList;
-        m_currentContactPtr = m_contactList.getPresentContact(m_currentTrajectoryTime);
+        if (!(activeContact->pose - activeContactNewList->pose).coeffs().isZero(tolerance))
+        {
+            log()->error("{} The pose of the contact in the new contact list is different from the "
+                         "pose of the contact in the original contact list. Given the contact "
+                         "lists and the time instant the contacts are active."
+                         "Original contact: {}, new contact: {}. Error {}. Current time {}.",
+                         logPrefix,
+                         activeContact->pose.coeffs().transpose(),
+                         activeContactNewList->pose.coeffs().transpose(),
+                         (activeContactNewList->pose - activeContact->pose).coeffs().transpose(),
+                         toMilliseconds(m_currentTrajectoryTime));
+            return false;
+        }
```

Comment:
Enabling this may lead to undesired effects we didn't take into consideration. Sincerely speaking. At this stage, I don't know what we should do. Probably recreate a new SwingFootPlanner. I think we can understand this limitation as soon as we encounter this problem.

@paolo-viceconte do you have any idea on this?

Reply:
Ok sounds good, let's consider this problem postponed